internal/discovery/discovery.go

```diff
@@ -65,7 +72,15 @@ func (c *Client) Discover(ctx context.Context) (endpoints []endpoint.Endpoint, e
 		return nil, xerrors.WithStackTrace(err)
 	}
 
-	response, err = c.service.ListEndpoints(ctx, &request)
+	client, cc, err := c.client(ctx)
+	if err != nil {
+		return nil, xerrors.WithStackTrace(err)
+	}
+	defer func() {
+		_ = cc.Close()
```

Comment:
here and in other similar places maybe it's better to make a wrapper for close() to apply a simpler defer form?

Reply:
simple form have linter issue (errcheck)

Reply:
that's why I suggested making a wrapper for close()
something like:

```go
func (c someType) silentClose {
  _ = c.Close()
}

...
// somewhere in the code
defer cc.silentClose()
```